datafusion/core/src/logical_plan/builder.rs

```diff
@@ -17,7 +17,6 @@
 
 //! This module provides a builder for creating LogicalPlans
 
-use crate::datasource::TableProvider;
```

Comment:
This was the main objective - removing one of the final dependencies from `LogicalPlanBuilder` to the "core" `datafusion` crate.